core/src/main/scala/no/nrk/bigquery/BQTableLike.scala

```diff
@@ -90,6 +93,28 @@ case class BQTableRef[+P](
 
   override def withTableType[PP](tpe: BQPartitionType[PP]): BQTableRef[PP] =
     BQTableRef(tableId, tpe)
+
+  def asFragment: BQSqlFrag = tableId.asFragment
+}
+
+case class BQAppliedTableValuedFunction[+P, N <: Nat](
+    tvf: TVF[P, N],
+    args: Sized[IndexedSeq[BQSqlFrag], N]
+) extends BQTableLike[P] {
+  override def tableId: BQTableId =
+    BQTableId(tvf.name.dataset, tvf.name.name.value)
```

Comment:
Not happy with this is a `BQTableId`. We have talked about having a reference type and I think that will be more intuitive here.

Reply:
Future work in https://github.com/nrkno/bigquery-scala/pull/202